src/main/java/org/hydev/mcpm/client/commands/entries/InfoController.java

```diff
@@ -0,0 +1,83 @@
+package org.hydev.mcpm.client.commands.entries;
+
+import com.google.common.collect.Streams;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.text.WordUtils;
+import org.hydev.mcpm.client.database.PluginTracker;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Controller for the info use case
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-22
+ */
+public record InfoController(PluginTracker tracker)
+{
+    private static final int LEN_LEFT = 12;
+    private static final int LEN_RIGHT = 60;
+
+    /**
+     * Format a kv pair if value isn't null, return empty string if value is null
+     *
+     * @param key Key
+     * @param rawValue Value
+     * @return Formatted pair
+     */
+    private static String formatPair(String key, @Nullable Object rawValue)
```

Comment:
Factor out Non-Null?

Reply:
wdym? I'm taking in nullable raw value because then I don't need to check if `plugin.something()` exists one by one

Reply:
This isn't necessarily bad but it does mean this method is responsible for unwrapping null. When a null value can completely switch the behaviour of a method like this, I like to move the non-null handling logic out into another method, and have another method that accepts null switch between the behaviours. I feel like this makes it harder to "opt-into" default null handling that you might not want/expect.

This method is private so it's your call, but I would still recommend moving the null-handling behaviour into its own method.

Reply:
I'm not going to get mad if you don't do it but...

> It would make it really verbose otherwise so I'll keep it like this...

No it wouldn't?

```java
private static String formatPairMaybe(String key, @Nullable Object rawValue) { 
  if (rawValue == null) {
    return formatPair(key, rawValue);
  } else {
    return "";
  }
}
```

I'm not asking you to do any programming gymnastics here...

Reply:
Oh ok right